internal/reconciliations/istio/configuration.go

```diff
@@ -60,14 +60,19 @@ func EvaluateIstioCRChanges(istioCR operatorv1alpha1.Istio, istioTag string) (tr
 		trigger = trigger | VersionUpdate
 	}
 
-	lastAppliedNotNil := lastAppliedConfig.Config.NumTrustedProxies != nil
-	newNotNil := istioCR.Spec.Config.NumTrustedProxies != nil
+	if nilChange(lastAppliedConfig.Components, istioCR.Spec.Components) {
+		return trigger | ConfigurationUpdate, nil
```

Comment:
I think we should simplify the IstioCRChange a lot, since we only care about two information -> `install with given configuration` or `delete`. All this extra states that are hidden in the int are not relevant right now and therefore we should remove it.